cha/design/sync.tex

```diff
@@ -0,0 +1,97 @@
+GPS satellites contain multiple atomic clocks.
+These clocks are monitored and controlled to keep them synchronized with the UTC standard.
+This provides us with an accurate clock to synchronize towards.\cite{gpsclock}
+After testing this, by developing an application to retrieve time from GPS, it was clear to us that the GPS hardware in our phones are limited to 1Hz updates and do not provide millisecond precision, which makes us unable to use GPS as a solution.
+
+\subsubsection{Audio}\mnote{Can go into more detail about how we would solve the difficulties in this, but i do not think it is worth}
+Another possible method is to use the audio to synchronize.
+This would have every device attempt to synchronize based on their perception.
+This method introduces several issues to be addressed, first of all there would be a ramp up time for the devices to figure out synchronization.
+Furthermore adding more devices to the network introduces more complexity as each introduces a new audio source, and with each device figuring out their own synchronization it is unlikely that the devices are synchronization towards the same source.
+While this may not have an effect on humans given only a few milliseconds of variance, due to psychoacoustic effects, it complicates the synchronization of the devices and our possible manipulation.
+
+\subsubsection{Light Cues}
+Alternatively we can use light rather than audio to synchronize.
+This would also allow devices to respond as to achieve synchronization better.
+The idea would be that a device uses the flashlight, once observed by another device it flashes back.
+Using Einstein Synchronization we can use the respective times collected from this, to synchronize the two devices \cite{einstein}.
+This has the significant disadvantage that the facing of the phones becomes important for synchronization, as the phones must have their flashes and cameras facing the synchronization target.
+%https://en.wikipedia.org/wiki/Einstein_synchronisation
+
+\subsubsection{\acl{NTP}}
+The last two options we are considering are to use existing protocols used for synchronization, one of which is \ac{NTP}.
+\ac{NTP} works by using a hierarchical system with high--precision atomic clocks at the highest level, each level is then synchronized towards devices from the higher level.
+\ac{NTP} polls multiple servers in the hierarchy and uses this to determine an actual time.
+Essentially the \ac{NTP} servers we will be polling act like a clock master, as such even the master of the network will have an offset.
+This also produces a hybrid solution between our proposed local centralized and remote centralized solutions, mentioned in \cnameref{sec:external_architectures}, by making the clock remote centralized, but maintaining all other control in a local network.
+The idea here would be for each device to keep track of their own \ac{NTP} offset, and then have then play in accordance with the time acquired from \ac{NTP}. 

```

Comment:
"then play in accordance" det er jo ikke kun play, men nærmere at udføre kommandoer på det aftalte tidspunkt.